src/toil/job.py

```diff
@@ -1176,8 +1192,11 @@ def readGlobalFile(self, fileStoreID, userPath=None, cache=True, mutable=None):
                     logger.debug('CACHE: Cache miss on file with ID \'%s\'.' % fileStoreID)
                     if fileIsLocal and cache:
                         # If caching of the downloaded file is desired, First create the .partial
-                        # file so other jobs know not to redundantly download the same file.
-                        open(harbingerFileName, 'w').close()  # This emulates the system 'touch'
+                        # file so other jobs know not to redundantly download the same file.  Write
+                        # the PID of this process into the file so other jobs know who is carrying
+                        # out the download.
+                        with open(harbingerFileName, 'w') as harbingerFile:
```

Comment:
Yes, you need to create this atomically. We can't just reason "the lock is held, so I don't need to worry about atomicity" like we did earlier today.